testsuite/tests/typing-modules-bugs/pr7726_bad.ml

```diff
@@ -0,0 +1,9 @@
+module type T = sig type t end
+
+module Fix(F:(T -> T)) = struct
+  module rec Fixed : T with type t = F(Fixed).t = F(Fixed)
+end
+
+module T = Fix(functor (X:sig type t end) -> struct type t = X.t option end)
+
+module T = Fix(functor (X:sig type t end) -> struct type t = X.t end)
```

Comment:
This test also fails on trunk (because of "Multiple definition of the module name T") which defeats its purpose.

Reply:
Well spotted.
I wonder wether dropping completely `ocamlc` tests for typing wouldn't be better; using `ocaml` avoids these pitfalls.

Reply:
Generally I think that that (except for tests that purposefully involve `.ml` and `.mli` files) using the expect-test format is superior for typing tests: it is more configurable, and (even more importantly in my opinion) it makes for more readable test files that express our intent better. Feel free to incrementally upgrade to expect-tests as you need to make changes to more tests.

Reply:
Indeed. The current test does not even check the error message at all, only that the program doesn't type-check.  In particular, it doesn't check that the problem is reported on the functor definition, not its use (this can be addressed by removing the functor application from the test, but obviously it would be better to use an expect test, or at least a test with the toplevel and a reference file).